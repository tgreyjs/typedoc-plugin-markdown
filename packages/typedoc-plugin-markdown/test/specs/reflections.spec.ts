import * as Handlebars from 'handlebars';

import { TestApp } from '../test-app';

describe(`Reflections:`, () => {
  let testApp: TestApp;
  let reflectionTemplate: Handlebars.TemplateDelegate;

  beforeAll(() => {
    testApp = new TestApp(['reflections.ts']);
  });

  describe(`(header)`, () => {
    beforeEach(async () => {
      await testApp.bootstrap({
        hideBreadcrumbs: false,
        hidePageTitle: true,
      });
      TestApp.stubPartials(['comment', 'member.signature', 'members']);
      TestApp.stubHelpers(['toc', 'breadcrumbs', 'hierarchy']);
      reflectionTemplate = TestApp.getTemplate('reflection');
    });
    test(`should compile template with breadcrumbs and without title`, () => {
      expect(
        TestApp.compileTemplate(reflectionTemplate, {
          model: testApp.project.children[0],
          project: testApp.project,
        }),
      ).toMatchSnapshot();
    });
  });

  describe(`(template)`, () => {
    beforeEach(async () => {
      await testApp.bootstrap({
        hideBreadcrumbs: true,
        hidePageTitle: false,
      });
      TestApp.stubPartials(['index', 'comment', 'member.signature', 'members']);
      TestApp.stubHelpers(['breadcrumbs', 'hierarchy']);
      reflectionTemplate = TestApp.getTemplate('reflection');
    });

    test(`should compile module with breadcrumbs and project title`, () => {
      expect(
        TestApp.compileTemplate(reflectionTemplate, {
          model: testApp.project.children[0],
          project: testApp.project,
        }),
      ).toMatchSnapshot();
    });

    test(`should compile a callable reflection`, () => {
      expect(
        TestApp.compileTemplate(reflectionTemplate, {
          model: testApp.findReflection('CallableReflection'),
          project: testApp.project,
        }),
      ).toMatchSnapshot();
    });

    test(`should compile an indexable reflection`, () => {
      expect(
        TestApp.compileTemplate(reflectionTemplate, {
          model: testApp.findReflection('IndexableReflection'),
          project: testApp.project,
        }),
      ).toMatchSnapshot();
    });

    test(`should compile implemented class`, () => {
      expect(
        TestApp.compileTemplate(reflectionTemplate, {
          model: testApp.findReflection('ImplementedClass'),
          project: testApp.project,
        }),
      ).toMatchSnapshot();
    });

    test(`should compile Enum`, () => {
      expect(
        TestApp.compileTemplate(reflectionTemplate, {
          model: testApp.findReflection('EnumReflection'),
          project: testApp.project,
        }),
      ).toMatchSnapshot();
    });
  });
});
