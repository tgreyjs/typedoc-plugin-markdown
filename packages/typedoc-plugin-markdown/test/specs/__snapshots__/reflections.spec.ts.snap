// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reflections: (header) should compile template with breadcrumbs and without title 1`] = `
"[helper: breadcrumbs]

[helper: toc]

[partial: members]
"
`;

exports[`Reflections: (template) should compile Enum 1`] = `
"# Enumeration: EnumReflection

[helper: toc]

[partial: members]
"
`;

exports[`Reflections: (template) should compile a callable reflection 1`] = `
"# Interface: CallableReflection

## Callable

### CallableReflection

[partial: member.signature]

[helper: toc]

[partial: members]
"
`;

exports[`Reflections: (template) should compile an indexable reflection 1`] = `
"# Interface: IndexableReflection

## Indexable

â–ª [index: \`number\`]: \`string\`

[partial: comment]

[helper: toc]

[partial: members]
"
`;

exports[`Reflections: (template) should compile implemented class 1`] = `
"# Class: ImplementedClass

## Implements

- [ReflectionClass](../classes/reflectionclass.md)

[helper: toc]

[partial: members]
"
`;

exports[`Reflections: (template) should compile module with breadcrumbs and project title 1`] = `
"# Enumeration: EnumReflection

[helper: toc]

[partial: members]
"
`;
